### PR TITLE
Travis improve

### PR DESCRIPTION
**Pull request by soullivaneuh**
@rande need to check if `GITHUB_OAUTH_TOKEN` is enabled on Travis. Don't have enough right on this project.

Note: PR needed for php-cs review.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -5,30 +5,48 @@ php:
...
   - hhvm
-  - nightly
+
+sudo: false
+
+cache:
+  directories:
+    - $HOME/.composer/cache
+    - $HOME/.cache/pip
+    - vendor
```
Comment (greg0ire) [inline .travis.yml:17]: What does this do ? Not sure we want to cache that (especially if using `--prefer-lowest`)

Comment (soullivaneuh) [inline .travis.yml:17]: Caches:
- pip global vendors
- composer global vendor
- project vendor

Project vendor would be updated but just the necessary.

This Travis config was adopted and patched on many sonata project.

cc @rande

File: .travis.yml [hunk 2]
```diff
@@ -5,30 +5,48 @@ php:
...
   allow_failures:
+    - php: 7.0
     - php: hhvm
-    - env: SYMFONY_VERSION=2.7.*@dev
-    - php: nightly
+    - env: SYMFONY_VERSION=2.8.*@dev
+    - env: SYMFONY_VERSION="3.0.x-dev as 2.8"
 
 before_script:
+  - mkdir -p ~/.phpenv/versions/$(phpenv version-name)/etc/conf.d && echo "memory_limit=-1" >> ~/.phpenv/versions/$(phpenv version-name)/etc/conf.d/travis.ini
```
Comment (greg0ire) [inline .travis.yml:45]: Please use long options

File: .travis.yml [hunk 3]
```diff
@@ -5,30 +5,48 @@ php:
...
   - composer selfupdate
-  - composer require symfony/symfony:${SYMFONY_VERSION} --no-update
-  - composer update --prefer-source $COMPOSER_FLAGS
+  - composer config -q github-oauth.github.com $GITHUB_OAUTH_TOKEN
```
Comment (greg0ire) [inline .travis.yml:47]: long options here too please

File: .travis.yml (hunk 2 above)
Comment (greg0ire) [inline .travis.yml:45]: This line is too long

File: .travis.yml [hunk 4]
```diff
@@ -5,30 +5,48 @@ php:
...
+  - if [ "$SYMFONY_VERSION" = "2.8.*@dev" ] || [ "$SYMFONY_VERSION" = "3.0.x-dev as 2.8" ]; then SYMFONY_DEPRECATIONS_HELPER=strict; fi;
+  - if [ "$SYMFONY_VERSION" != "" ]; then composer require "symfony/symfony:${SYMFONY_VERSION}" --no-update; fi;
```
Comment (greg0ire) [inline .travis.yml:49]: these lines are too long

**Comment by soullivaneuh** [comment]
@greg0ire What did you mean by long options?

BTW this config is actually working on other bundle, please see: https://github.com/sonata-project/SonataAdminBundle/blob/master/.travis.yml

File: .travis.yml (hunk 1 above)
Comment (greg0ire) [inline .travis.yml:17]: Ok I get it, thanks for the explanation, it is a good to get faster builds :)

**Comment by soullivaneuh** [comment]
I prefer not customize it specifically for this bundle as a work for a "common configuration" will be done.

Maybe we could discuss about optimization here.

**Comment by greg0ire** [comment]
I mean `mkdir --parents`, for instance, and whatever  `-q` means. More noob-friendly.

**Comment by greg0ire** [comment]
Why does the build fail ? I don't get it, I just pushed another modification to fix a typo in a filename and the build for that modification works fine…

**Comment by soullivaneuh** [comment]
@greg0ire don't know for the moment. Will investigate.

**Comment by rande** [comment]
@greg0ire `-q` means quiet so the key is not displayed on travis

**Comment by rande** [comment]
@greg0ire just for information @Soullivaneuh is helping a lot to clean sonata, fix deprecated errors and dealing with issues/PRs. This PR just try to make the CI like others bundles.

**Comment by greg0ire** [comment]
Ok. Please keep in mind my remarks when discussing the common configuration though. Namely : 
- When writing shell scripts, always use long options when available, they are easier to understand. Short options are for typing commands directly in your shell
- Long line should be avoided in every language / configuration if possible. 

I will merge this as soon as the build passes.

**Comment by soullivaneuh** [comment]
Can't find why it's failing: https://travis-ci.org/sonata-project/SonataAdminSearchBundle/jobs/67512304

@rande @greg0ire any idea?

**Comment by greg0ire** [comment]
Yes : 

```
Installing matthiasnoback/symfony-dependency-injection-test (v0.7.3)

    Downloading: Connecting...    Failed to download matthiasnoback/symfony-dependency-injection-test from dist: Could not authenticate against github.com

    Now trying to download from source
```

**Comment by soullivaneuh** [comment]
@greg0ire This is another know problem which will be resolved when integrated on master.

I'm talking of the previously linked one: https://travis-ci.org/sonata-project/SonataAdminSearchBundle/jobs/67512304

**Comment by greg0ire** [comment]
You are talking about this error, aren't you ?

> Class 'Matthias\SymfonyDependencyInjectionTest\PhpUnit\AbstractCompilerPassTestCase' not found in /home/travis/build/sonata-project/SonataAdminSearchBundle/Tests/DependencyInjection/Compiler/DatagridBuilderInjectionCompilerPassTest.php on line 11

**Comment by soullivaneuh** [comment]
Exactly yes. I got the same error on my local but I can't figure out why because the class exists and namespace seems to be OK.

**Comment by soullivaneuh** [comment]
BTW, I got the same error on master.

**Comment by greg0ire** [comment]
Oh I thought it failed to download but it fact it did.

Maybe try adding this to the build : `ls -R vendor/matthiasnoback && cat vendor/matthiasnoback/symfony-dependency-injection-test/PhpUnit/AbstractCompilerPassTestCase.php`

This will help us now whether the download was successful or not

**Comment by greg0ire** [comment]
Oh you have the same in local. Nevermind what I said then.

**Comment by soullivaneuh** [comment]
Well...

``` bash
$ ./vendor/bin/phpunit --version
PHPUnit 4.7.4 by Sebastian Bergmann and contributors.
$ ./vendor/bin/phpunit 
PHPUnit 4.7.4 by Sebastian Bergmann and contributors.

..............

Time: 194 ms, Memory: 10.25Mb

OK (14 tests, 18 assertions)
```

``` bash
$ ~/.composer/vendor/bin/phpunit --version
PHPUnit 4.7.4 by Sebastian Bergmann and contributors.

$ ~/.composer/vendor/bin/phpunit -v
PHP Fatal error:  Class 'Matthias\SymfonyDependencyInjectionTest\PhpUnit\AbstractCompilerPassTestCase' not found in /home/sullivan/projects/fork/SonataAdminSearchBundle/Tests/DependencyInjection/Compiler/DatagridBuilderInjectionCompilerPassTest.php on line 11
PHP Stack trace:
PHP   1. {main}() /home/sullivan/.composer/vendor/phpunit/phpunit/phpunit:0
PHP   2. PHPUnit_TextUI_Command::main() /home/sullivan/.composer/vendor/phpunit/phpunit/phpunit:36
PHP   3. PHPUnit_TextUI_Command->run() /home/sullivan/.composer/vendor/phpunit/phpunit/src/TextUI/Command.php:103
PHP   4. PHPUnit_TextUI_Command->handleArguments() /home/sullivan/.composer/vendor/phpunit/phpunit/src/TextUI/Command.php:113
PHP   5. PHPUnit_Util_Configuration->getTestSuiteConfiguration() /home/sullivan/.composer/vendor/phpunit/phpunit/src/TextUI/Command.php:626
PHP   6. PHPUnit_Util_Configuration->getTestSuite() /home/sullivan/.composer/vendor/phpunit/phpunit/src/Util/Configuration.php:851
PHP   7. PHPUnit_Framework_TestSuite->addTestFiles() /home/sullivan/.composer/vendor/phpunit/phpunit/src/Util/Configuration.php:938
PHP   8. PHPUnit_Framework_TestSuite->addTestFile() /home/sullivan/.composer/vendor/phpunit/phpunit/src/Framework/TestSuite.php:411
PHP   9. PHPUnit_Util_Fileloader::checkAndLoad() /home/sullivan/.composer/vendor/phpunit/phpunit/src/Framework/TestSuite.php:339
PHP  10. PHPUnit_Util_Fileloader::load() /home/sullivan/.composer/vendor/phpunit/phpunit/src/Util/Fileloader.php:40
PHP  11. include_once() /home/sullivan/.composer/vendor/phpunit/phpunit/src/Util/Fileloader.php:56

```

So yes, calling phpunit outside vendor cause this.

But update Travis in consequence is not a fix, just a workaround.

I'll talk with Matthias about that.

**Comment by greg0ire** [comment]
Wow, that's really, really strange…

**Comment by soullivaneuh** [comment]
> Wow, that's really, really strange…

Hell yes. :-)

**Comment by soullivaneuh** [comment]
@greg0ire Yes if you want. I'll do this at the end when phpunit issue will be fixed. :+1:

**Comment by soullivaneuh** [comment]
Commit renamed and test passed (finally!), going to merge it.

**Comment by greg0ire** [comment]
Nice, thanks @Soullivaneuh !

**Comment by soullivaneuh** [comment]
:+1:

**Pull request merged**